api/src/db/models/opportunity_models.py

```diff
@@ -70,7 +70,10 @@ class Opportunity(ApiSchemaTable, TimestampMixin):
     )
 
     agency_record: Mapped[Agency | None] = relationship(
-        Agency, primaryjoin="Opportunity.agency == foreign(Agency.agency_code)", uselist=False
+        Agency,
+        primaryjoin="Opportunity.agency == foreign(Agency.agency_code)",
+        uselist=False,
+        viewonly=True,
```

Comment:
The ELT job is failing in dev because of this:
```
"(psycopg.errors.NotNullViolation) null value in column \"agency_code\" of relation \"agency\" violates not-null constraint\nDETAIL:  Failing row contains (914, MicroHealth LLC, null, MH, 00.00, 3, 914, f, MH, MicroHealth LLC, MicroHealth LLC, f, f, t, f, f, f, 2021-09-17 16:06:49+00, 2024-10-23 15:53:13.128999+00, null).\n[SQL: UPDATE api.agency SET agency_code=%(agency_code)s::VARCHAR, updated_at=%(updated_at)s::TIMESTAMP WITH TIME ZONE WHERE api.agency.agency_id = %(api_agency_agency_id)s::BIGINT]\n[SQL parameters hidden due to hide_parameters=True]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)",
```

We don't even have a true foreign key here, it's a bit odd that this is the default behavior, but this fixes it.